chia/wallet/trading/offer.py

```diff
@@ -242,15 +253,20 @@ def aggregate(cls, offers: List["Offer"]) -> "Offer":
                 raise ValueError("The aggregated offers overlap inputs")
 
             # Next, do the aggregation
-            for tail, payments in offer.requested_payments.items():
-                if tail in total_requested_payments:
-                    total_requested_payments[tail].extend(payments)
+            for asset_id, payments in offer.requested_payments.items():
+                if asset_id in total_requested_payments:
+                    total_requested_payments[asset_id].extend(payments)
                 else:
-                    total_requested_payments[tail] = payments
+                    total_requested_payments[asset_id] = payments
+
+            for key, value in offer.driver_dict.items():
+                if key in total_driver_dict and total_driver_dict[key] != value:
+                    raise ValueError(f"The offers to aggregate disagree on the drivers for {key.hex()}")
 
             total_bundle = SpendBundle.aggregate([total_bundle, offer.bundle])
+            total_driver_dict = {**total_driver_dict, **offer.driver_dict}
```

Comment:
It's just a (at least imo) simpler statement and you update the dict instead of creating a new one each time. Like you would use `dict["new_key"] = new_value` instead of `dict = {**dict, "new_key": new_value}`.